src/lib/cloud-provider.js

```diff
@@ -57,12 +58,31 @@ class CloudProvider {
     }
 
     onOpen () {
+        this.connectionAttempts = 1; // Reset because we successfully connected
         this.writeToServer('handshake');
         log.info(`Successfully connected to clouddata server.`);
     }
 
     onClose () {
         log.info(`Closed connection to websocket`);
+        const exponentialTimeout = (Math.pow(2, this.connectionAttempts) - 1) * 1000;
+        const randomizedTimeout = this.randomizeDuration(exponentialTimeout);
+        this.setTimeout(this.reconnectNow.bind(this), randomizedTimeout);
+    }
+
+    reconnectNow () {
+        // Max connection attempts at 5, so timeout will max out in range [0, 31s]
+        this.connectionAttempts = Math.min(this.connectionAttempts + 1, 5);
+        this.openConnection();
+    }
+
+    randomizeDuration (t) {
+        return Math.random() * t;
+    }
+
+    setTimeout (fn, time) {
+        log.info(`Reconnecting in ${time}ms, attempt ${this.connectionAttempts}`);
```

Comment:
The attempt number in this log message should probably always be increasing, regardless of the fact that we capped out in the decay calculation...